### ISSUE TITLE
 3.2.3 Consistent Navigation -scope of "relative order", and nature of this order: source order, visual presentation, both?

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag#2133]
the current wording for the SC https://www.w3.org/TR/WCAG21/#consistent-navigation - including the clarification of the term "same relative order" - still leaves quite a few vague gaps.

does the "same relative order" only apply to the navigation mechanisms themselves amongst each other (which is what the definition of the term hints at https://www.w3.org/TR/WCAG21/#dfn-same-relative-order? just making sure that links/buttons used for navigation are in the same relative order? or does this also include a broader "and the order that those navigation mechanisms have with the rest of the document/page components"? for instance, if i have a `<nav>` that is always the same (so no issue of relative order for the navigation elements inside it), but on some pages it comes before the `<main>`, while on others it inexplicably comes after the `<main>`, is that a failure or 3.2.3 because the relative order of the whole navigation compared to the main content area is not the same?

further, the understanding document https://www.w3.org/WAI/WCAG21/Understanding/consistent-navigation.html starts with

> The intent of this Success Criterion is to encourage the use of consistent presentation and layout

but later talks about the advantage for blind users. So, by implication, this isn't just about visual presentation/layout, but also source/focus order? if so, this should be clarified.

**Comment by detlevhfischer** [comment]
My guess is that in general, having navigation areas move around relative to content and other navigation areas would be as confusing as a change of relative order inside navigation elements. I have always assumed that both aspects are covered. The questions I've had with the SC are more regarding potentially valid exceptions where (a part of) the standard navigation is missing. For processes this would be acceptable, but would a search results page, for example, be considered part of a process, so that not having the full navigation options here can be accepted? Another case is having subsections on a site where navigation changes in some ways, but is otherwise consistent within that subsection.

**Comment by alastc** [comment]
"Navigational mechanisms" are the object of the SC text, so I've always thought it meant that navs should be in the same order. It is also good to keep the layout consistent, but I haven't considered that in scope for pass/fail.

Regarding code/visual layout, I've interpreted that as: the order users can encounter them, so both.

So for a keyboard user it would be the tab order, for a screenreader user, the tab and content order, and visually the layout order (primarily vertically). Generally those orders line up unless they use some odd layout methods, which is a good trigger for checking this SC.

I don't think your nav/main example would fail because the main isn't a navigation mechanism.

**Comment by patrickhlauke** [comment]
> I don't think your nav/main example would fail because the main isn't a navigation mechanism.

thing is though, the main part could contain other focusable/navigable things (links etc), so overall (on a whole-page level) the navigational elements don't occur in the same relative order (for the overall page).

i do think if we get agreement, this should be clarified (that basically that this is either just about the relative order "within the navigational mechanisms themselves", or their order not just internally, but in the context of the overall page/document structure). plus clarification that this "relative order" is both about visual layout and about structural/reading order/focus order

**Comment by mraccess77** [comment]
[F66 ](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/F66) seems to say that links within a navigation block itself need to remain in order.

We all need to come to some agreement whether blocks as well as links within the blocks are covered.

**Comment by patrickhlauke** [comment]
@mraccess77 the individual links in that case count as individual "navigation mechanisms", so their relative order among each other matters/falls under the SC

**Comment by yatil** [comment]
This is interesting to me, as I would have not identified links inside a navigation as “navigational mechanisms” themselves. (The rarity of this happening in modern, CMS-based websites might also have contributed to not being aware of it.) 

I think in lieu of defining “navigational mechanisms” in WCAG proper, a clarification in the Understanding would be appreciated.
